receiver/vmmetricsreceiver/vm_metrics_collector.go

```diff
@@ -84,8 +88,25 @@ func NewVMMetricsCollector(si time.Duration, mountPoint, processMountPoint, pref
 	return vmc, nil
 }
 
+func detectResource() {
+	res, err := auto.Detect(context.Background())
+	if err != nil {
+		rsc = nil
```

Comment:
I didn't see `auto.Detect` code but what are the implications of this silent failure? when can it happen? is it worth having svc running with this broken?

Reply:
good question. I think if there is err it should panic. Some backends may not even plot the graph if resource is not correct. Let me change it.